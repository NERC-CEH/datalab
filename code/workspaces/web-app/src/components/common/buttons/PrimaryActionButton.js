import React from 'react';
import Button from '@material-ui/core/Button';

const PrimaryActionButton = ({ children, variant, color, ...rest }) => (
  <Button
    variant="outlined"
    color="primary"
    {...rest}
  >
    {children}
  </Button>
);

export default PrimaryActionButton;
