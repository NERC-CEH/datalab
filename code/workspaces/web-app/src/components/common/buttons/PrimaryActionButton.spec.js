import React from 'react';
import { shallow } from 'enzyme';
import PrimaryActionButton from './PrimaryActionButton';

function shallowRender(props) {
  return shallow(<PrimaryActionButton {...props} />);
}

describe('PrimaryActionButton', () => {
  it('overrides variant and color props', () => {
    expect(shallowRender({ variant: 'contained', color: 'secondary' })).toMatchSnapshot();
  });
});
