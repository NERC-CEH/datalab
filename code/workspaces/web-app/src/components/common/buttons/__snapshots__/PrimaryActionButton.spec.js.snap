// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryActionButton overrides variant and color props 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-o0h1hg-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <div>
      Test Child
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
